bluesky/spec_api.py

```diff
@@ -357,3 +355,162 @@ def tw(motor, step, time=None, *, md=None):
+
+    Parameters
+    ----------
+    x_motor : object
+        any 'setable' object (motor, temp controller, etc.)
+    y_motor : object
+        any 'setable' object (motor, temp controller, etc.)
+    x_range : float
+        x range of spiral
+    y_range : float
+        y range of spiral
+    dr : float
+        delta radius
+    factor : float
+        radius gets divided by this
+    time : float, optional
+        applied to any detectors that have a `count_time` setting
+    per_step : callable, optional
+        hook for cutomizing action of inner loop (messages per step)
+        See docstring of bluesky.plans.one_nd_step (the default) for
+        details.
+    md : dict, optional
+        metadata
+    '''
+    plan = afermat(x_motor, y_motor, x_motor.position, y_motor.position,
+                   x_range, y_range, dr, factor, time=time, per_step=per_step,
+                   md=md)
+    plan = plans.reset_positions(plan)  # return motors to starting pos
+    return [plan]
+
+
+@planify
+def aspiral(x_motor, y_motor, x_start, y_start, x_range, y_range, dr, nth,
+            time=None, *, per_step=None, md=None):
+    '''Spiral scan, centered around (x_start, y_start)
+
+    Parameters
+    ----------
+    x_motor : object
+        any 'setable' object (motor, temp controller, etc.)
+    y_motor : object
+        any 'setable' object (motor, temp controller, etc.)
+    x_range : float
+        X range, in engineering units
+    y_range : float
+        Y range, in engineering units
+    dr : float
+        Delta radius, in engineering units
+    nth : float
+        Number of theta steps
+    time : float, optional
+        applied to any detectors that have a `count_time` setting
+    per_step : callable, optional
+        hook for cutomizing action of inner loop (messages per step)
+        See docstring of bluesky.plans.one_nd_step (the default) for
+        details.
+    md : dict, optional
+        metadata
+    '''
+    subs = {'all': [LiveTable([x_motor, y_motor, gs.PLOT_Y] + gs.TABLE_COLS),
+                    ]}
+
+    plan_stack = deque()
+    with plans.subs_context(plan_stack, subs):
+        plan = plans.spiral(gs.DETS, x_motor, y_motor, x_start,
+                            y_start, x_range, y_range, dr,
+                            nth, per_step=per_step, md=md)
+        plan = configure_count_time(plan, time)
+        plan_stack.append(plan)
+    return plan_stack
+
+
+@planify
+def spiral(x_motor, y_motor, x_range, y_range, dr, factor, time=None, *,
+           per_step=None, md=None):
+    '''Relative spiral scan
+
+    Parameters
+    ----------
+    x_motor : object
+        any 'setable' object (motor, temp controller, etc.)
+    y_motor : object
+        any 'setable' object (motor, temp controller, etc.)
+    x_range : float
+        X range, in engineering units
+    y_range : float
+        Y range, in engineering units
+    dr : float
+        Delta radius, in engineering units
+    nth : float
```

Comment:
I think you meant to leave this on the next function down - it was named factor, should have been nth. (correct me if I'm wrong)